### ISSUE TITLE
Error: listen EADDRINUSE 0.0.0.0:8000 but no other process is running

### ISSUE THREAD

**Issue by crucialfelix** [gatsbyjs/gatsby#334]
This is a bit curious.

1. Start up another webpack project that is serving on 0.0.0.0:8000
2. Try to run gatsby. It correctly complains of EADDRINUSE and hangs. 
3. Ctrl-c gatsby
4. Ctrl-c the other webpack
5. Try to run gatsby again. Still complains of EADDRINUSE and hangs

Checking htop there are no other processes running. But something (perhaps gatsby at step 2) is still hanging onto this port.

The only thing that works is to restart the whole computer.

```
> gatsby develop

{ [Error: listen EADDRINUSE 0.0.0.0:8000]
  code: 'EADDRINUSE',
  errno: 'EADDRINUSE',
  syscall: 'listen',
  address: '0.0.0.0',
  port: 8000 }
Listening at: http://0.0.0.0:8000
^C
```

Notice how it says it is listening, even though it got an EADDRINUSE error.

**Comment by KyleAMathews** [comment]
Thanks for the error report! I agree this is a sucky UX + you're right, the dev server should *not* start listening if there's an error.

I just created a PR to fix this #335. In the above scenario, Gatsby now outputs an error message like the following then quits.

```
We were unable to start Gatsby on port 8000 as there's already a process
listening on that port (PID: 43337). You can either use a different port
(e.g. gatsby develop --port 8001) or stop the process already listening
on your desired port.
```

**Comment by NogsMPLS** [comment]
+1000000 for including PID in the error message. bookmarking this so i can come back to it later for my own stuff.

**Comment by crucialfelix** [comment]
Sounds good.  Thanks !

On Thu, Jun 16, 2016 at 8:43 PM Kyle Mathews <notifications@github.com>
wrote:

> Thanks for the error report! I agree this is a sucky UX + you're right,
> the dev server should *not* start listening if there's an error.
>
> I just created a PR to fix this #335
> <https://github.com/gatsbyjs/gatsby/pull/335>. In the above scenario,
> Gatsby now outputs an error message like the following then quits.
>
> We were unable to start Gatsby on port 8000 as there's already a process
> listening on that port (PID: 43337). You can either use a different port
> (e.g. gatsby develop --port 8001) or stop the process already listening
> on your desired port.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gatsbyjs/gatsby/issues/334#issuecomment-226576314>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AANWci0ZmDBtiQltcQjCjApnGfyTYCmTks5qMZlZgaJpZM4I3Ka1>
> .
>
-- 
https://twitter.com/crucialfelix
https://medium.com/@crucialfelix
http://www.mattermind.com/

**Comment by KyleAMathews** [comment]
Just published 0.11.5 🎉 🎉 🎉 

Thanks for raising the issue @crucialfelix! This has always kinda bothered me but your pushing crystalized a good solution. Cheers and happy Gatsbing.

**Comment by paulclifford** [comment]
Instead of restarting your computer, you can try this. At the terminal:

1) `ps` copy the PID of the gatsby develop server to your clipboard
2) `kill -HUP [paste PID here]`
